lib/consts.ml

```diff
@@ -21,3 +21,6 @@ let wait_for_xapi_timeout_seconds = 300.0
 
 (** We sleep for this many seconds before the next login attempt. *)
 let wait_for_xapi_retry_delay_seconds = 4.0
+
+(** Allow no more than this many client connections. *)
+let connection_limit = 16
```

Comment:
The counter logic looks good, but where does the limit 16 come from?

Reply:
Yep, whatever number we choose it's gonna be a bit arbitrary. @thomassa wanted to limit it to 127 originally. The important thing is that the maximum number of VBDs linked to dom0 is limited to 256, and xapi also needs to attach VBDs to dom0 to perform some operations. So we have to leave enough VBDs for dom0 & xapi to do its work.

Reply:
But I do think that we could increase it to a larger number.

Reply:
I'd rather be conservative and increase it in a second stage if it blocks partners. Let's ask some architects: @robhoes

Reply:
Hard to know. And we _could_ make it settable through a command-line option or a config file or something.
But what's the maxiumum number of connections that would make sense? With 16 they will almost certainly be slowing each other down, and it would be probably be faster to read the blocks one VDI at a time... or maybe multiple VDIs but only one connection at a time using each PBD.

Reply:
But the host could have multiple NICs though?

Reply:
I'd say let's make it 32 or 50?

Reply:
Ok, let's limit it to 16, as we discussed.